### PR TITLE
feat(las): preserve legacy LAZ streaming state

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

- Remove bounded geometric replay from streaming LAZ PDRF 0-5 decoding.
- Preserve arithmetic and item-predictor state across arbitrary input chunk boundaries.
- Keep memory bounded by feeding only the consumed prefix plus a record-aware lookahead.

## Changes

- Adds appendable input and available-point decoding to `LAZChunkDecoderCursor` for legacy interleaved LAZ chunks.
- Reuses one decoder cursor per file chunk instead of reconstructing and replaying previously emitted points.
- Feeds only newly available compressed bytes and advances the file reader by the exact consumed chunk length.
- Adds a 4,096-point PDRF 2 one-byte-feed stress test plus existing PDRF 3/4/5 parity and conformance coverage.
- Updates LAS streaming documentation and the remaining implementation roadmap.

## Performance

Identical PDRF 3 comprehensive streaming benchmark before and after:

- Throughput: 1.59M to 1.70M points/second, about 7% faster.
- Copied input: 3,350,873 bytes to 81,106 bytes, 97.6% less.
- Concatenations: 25 to 5, 80% fewer.
- PDRF 7 common-column streaming remains about 2.05M points/second.

## Validation

- `yarn build`
- `yarn build-workers`
- `yarn test-node` (351 passed, 6 skipped)
- `yarn test-headless` (2,693 passed, 38 skipped)
- focused LAS browser tests (23 passed)
- slow LAS streaming conformance (11 passed)
- `yarn test-audit` (0 violations)
- `yarn bench las`
- `yarn lint fix`
- website production build

### CODE REVIEW

File: modules/loader-utils/test/lib/laz/laz-chunk-decoder.spec.ts [hunk 1]
```diff
@@ -99,3 +101,72 @@ test('layered LAZ chunk framing is available before layer payloads', () => {
...
+  const compressed = encodeLAZChunk(rawPointData, metadata);
+  const expected = decodeLAZChunk(compressed, metadata);
+  const cursor = createLAZChunkDecoderCursor(new Uint8Array(0), metadata);
+  const actual = new Uint8Array(expected.byteLength);
+  let decodedPointCount = 0;
+  let firstOutputByteLength = 0;
+
+  for (let byteOffset = 0; byteOffset < compressed.byteLength; byteOffset++) {
+    const byteEnd = byteOffset + 1;
+    cursor.feed(compressed.subarray(byteOffset, byteEnd));
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/loader-utils/test/lib/laz/laz-chunk-decoder.spec.ts:122]: Addressed in 12362b8b93. The fast suite now keeps a 64-point boundary case, while the original 4,096-point byte-at-a-time high-entropy stress case runs in `laz-chunk-decoder.slow.spec.ts`. Focused fast and slow tests pass, and the test audit reports no violations.
